packages/amplify-provider-awscloudformation/src/disconnect-dependent-resources/utils.ts

```diff
@@ -129,6 +130,13 @@ const generateIterativeFuncDeploymentOp = async (
   const funcStack = await cfnClient
     .describeStackResources({ StackName: rootStackId, LogicalResourceId: `function${functionName}` })
     .promise();
+
+  if (!funcStack.StackResources || funcStack.StackResources.length === 0) {
+    throw new AmplifyFault('ResourceNotFoundFault', {
+      message: `Could not find function ${functionName} in root stack ${rootStackId}`,
+    });
+  }
```

Comment:
1. Do we know why this situation can happen? 
2. And what customers can do to unblock themselves?

Reply:
we do not know, I didn't find any information in our usage data to help, so this is a first step for us to get more information if it still happens so we can fix the root cause